src/Orleans/Logging/LoggerImpl.cs

```diff
@@ -0,0 +1,614 @@
+        {
+            get
+            {
+                if (useCustomSeverityLevel || (defaultCopiedCounter >= LogManager.defaultModificationCounter)) return severity;
+
+                severity = LogManager.GetDefaultSeverityForLog(logName, loggerType);
+                defaultCopiedCounter = LogManager.defaultModificationCounter;
+                return severity;
+            }
+        }
+
+        public override Logger GetLogger(string loggerName)
+        {
+            return LogManager.GetLogger(loggerName, loggerType);
+        }
+
+        /// <summary>
+        /// Set a new severity level for this Logger.
+        /// Log entries will be written if their severity is (logically) equal to or greater than this level.
+        /// </summary>
+        /// <param name="sev">New severity level to be used for filtering log messages.</param>
+        public void SetSeverityLevel(Severity sev)
+        {
+            severity = sev;
+            useCustomSeverityLevel = true;
+        }
+
+        /// <summary>
+        /// Constructs a Logger with the given name and type.
+        /// </summary>
+        /// <param name="source">The name of the source of log entries for this Logger.
+        /// Typically this is the full name of the class that is using this Logger.</param>
+        /// <param name="logType">The category of Logger to create.</param>
+        internal LoggerImpl(string source, LoggerType logType)
+        {
+            defaultCopiedCounter = -1;
+            logName = source;
+            loggerType = logType;
+            useCustomSeverityLevel = CheckForSeverityOverride();
+        }
+
+        internal bool MatchesPrefix(string prefix)
+        {
+            return logName.StartsWith(prefix, StringComparison.Ordinal)
+                || (loggerType + "." + logName).StartsWith(prefix, StringComparison.Ordinal);
+        }
+
+        internal bool CheckForSeverityOverride()
+        {
+            lock (LogManager.lockable)
+            {
+                if (LogManager.traceLevelOverrides.Count <= 0) return false;
+
+                foreach (var o in LogManager.traceLevelOverrides)
+                {
+                    if (!MatchesPrefix(o.Item1)) continue;
+
+                    severity = o.Item2;
+                    useCustomSeverityLevel = true;
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        //----------------------
+
+        /// <summary>
+        /// Writes a log entry at the Verbose severity level.
+        /// Verbose is suitable for debugging information that should usually not be logged in production.
+        /// Verbose is lower than Info.
+        /// </summary>
+        /// <param name="format">A standard format string, suitable for String.Format.</param>
+        /// <param name="args">Any arguments to the format string.</param>
+        public override void Verbose(string format, params object[] args)
```

Comment:
I think these methods should either be moved to the base Logger class or become extension methods for the Logger class. Users should not have to reference this internal concrete class just to be able to use this syntactic sugar.